rust/xaynet-server/src/state_machine/mod.rs

```diff
@@ -151,6 +152,9 @@ pub enum StateMachineError {
 
     #[error("the request could not be processed due to an internal error")]
     InternalError,
+
+    #[error("dependency failed: {0}")]
+    DependencyFailed(#[from] RedisError),
```

Comment:
Maybe we could just name this one `Redis` because it contains a redis error anyway